WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,18 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    self.rounds.each(&:destroy!)
```

Comment:
Oh, interesting. That didn't occur to me. Mostly because it's _really_ hard to fail at deleting things in SQL. It can happen with foreign keys in some situations. And probably due to write locks in concurrent transactions.

Reading up, what `destroy!` does is raise an exception if the `before_destroy` hook stops deletion. I wouldn't personally worry about that in my code, but if you want to be extra safe, I can respect that.